src/main/java/org/toilelibre/libe/curl/AfterResponse.java

```diff
@@ -27,7 +27,12 @@ static void handle (final CommandLine commandLine, final HttpResponse response)
 
     private static void writeTheResponseEntityInsideStream(FileOutputStream outputStream, HttpEntity httpEntity) {
         try {
-            outputStream.write (IOUtils.toByteArray (httpEntity.getContent (), (int) httpEntity.getContentLength ()));
+            if (httpEntity.getContentLength () >= 0) {
+                outputStream.write (IOUtils.toByteArray (httpEntity.getContent (), (int) httpEntity.getContentLength ()));
+            }
+            else {
```

Comment:
Goid catch, so sometimes the size cannot be read beforehand in case of chunk data, right ?

Reply:
Usually this is the case for any streaming data where the size is not known. This is used a lot. HttpClient is providing a ChunkedInputStream. And this should be handle transparently. The input stream returned is such an instance. So maybe the code can be simplified and the size is never needed and it can be read until read returns EOF like in my new method. Check on this for simple data and if the unit tests are working then this can be reduced to one line without the if else